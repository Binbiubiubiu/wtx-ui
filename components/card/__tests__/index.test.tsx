import React from 'react';
import { render } from 'enzyme';
import toJson from 'enzyme-to-json';
import Card, { InlineCard } from '..';

describe('<Card>', () => {
  it('should render <Card/> components', () => {
    const wrapper = render(
      <>
        <Card icon={12} title="摄像头类型(种)" content="6" style={{ marginBottom: 20 }} />
        <InlineCard title="消防联网设备总数(个)" content="300" />
      </>,
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should style output is not null', () => {
    expect(require('../style')).not.toBeNull();
  });

  it('should Card render ok', () => {
    const wrapper = render(
      <>
        <Card icon={<span className="test-icon">icon</span>} title="标题" content={6} />
        <Card
          icon={<span className="test-icon">icon</span>}
          title="标题"
          content={<span className="test-content"></span>}
        />
      </>,
    );

    expect(wrapper.find('.test-icon')).toHaveLength(2);
    expect(wrapper.find('.test-content')).toHaveLength(1);
    expect(
      wrapper
        .find('.wtx-card__title')
        .eq(0)
        .text(),
    ).toEqual('标题');
    expect(wrapper.find('.wtx-card__content').text()).toEqual('6');
    expect(wrapper.find('.led-num')).toHaveLength(1);
  });

  it('should Card render ok', () => {
    const wrapper = render(<InlineCard title="标题" content="300" />);

    expect(wrapper.find('.wtx-inline-card__title').text()).toEqual('标题');
    expect(wrapper.find('.wtx-inline-card__content').text()).toEqual('300');
    expect(wrapper.find('.led-num')).toHaveLength(1);
  });
});
