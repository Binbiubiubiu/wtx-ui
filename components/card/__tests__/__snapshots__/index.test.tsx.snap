// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card> should render <Card/> components 1`] = `
Array [
  <div
    class="wtx-card"
    style="margin-bottom:20px"
  >
    12
    <dl
      class="wtx-card__body"
    >
      <dt
        class="wtx-card__title"
      >
        摄像头类型(种)
      </dt>
      <dd
        class="wtx-card__content"
      >
        <span
          class="led-num"
        >
          6
        </span>
      </dd>
    </dl>
  </div>,
  <dl
    class="wtx-inline-card"
  >
    <dt
      class="wtx-inline-card__title"
    >
      消防联网设备总数(个)
    </dt>
    <dd
      class="wtx-inline-card__content led-num"
    >
      300
    </dd>
  </dl>,
]
`;
